linkerd/config/src/main/scala/io/buoyant/linkerd/config/types/ThriftProtocolDeserializer.scala

```diff
@@ -0,0 +1,16 @@
+package io.buoyant.linkerd.config.types
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.twitter.finagle.thrift.Protocols
+import io.buoyant.linkerd.config.ConfigDeserializer
+import org.apache.thrift.protocol.{TCompactProtocol, TProtocolFactory}
+
+class ThriftProtocolDeserializer extends ConfigDeserializer[TProtocolFactory] {
+  override def deserialize(jp: JsonParser, ctxt: DeserializationContext): TProtocolFactory = {
+    _parseString(jp, ctxt) match {
+      case "binary" => Protocols.binaryFactory()
+      case "compact" => new TCompactProtocol.Factory()
```

Comment:
We should probably add an explicit throw as the default case rather than relying on a MatchError here